File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector < pair < int , int > > pr ;
        int n = nums.size();
        for(int i = 0 ; i < n ; i++) pr.push_back({nums[i], i});
        sort(pr.begin(), pr.end()) ;
        int i = 0 , j = n - 1 ;
        while(i <= j)
        {
            int x = pr[i].first + pr[j].first ;
            if (x > target)
                j-- ;

            else if (x < target) i++ ;
            else break ;
        }
        vector <int> v(2) ;
        v[0] = pr[i].second ; v[1] = pr[j].second ;
        return v ;
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function comparing(a , b){
    return a.num - b.num;  
}
var twoSum = function(nums, target) {
    let newArr = nums.map((num , idx) => ({num , idx})) ; 
    
    newArr.sort(comparing) ; // in-place sorting
    let l = 0 , r = nums.length - 1 ; 

    while(l < r){
        let item = newArr[r].num + newArr[l].num ; 
        if(item > target)
            r--; 
        else if(item < target)
            l++ ; 

        else return [newArr[l].idx , newArr[r].idx] ; 
    } 
    
};